src/embeddings/embeddings_calculator.cc

```diff
@@ -137,6 +135,23 @@ class EmbeddingsCalculator : public CalculatorBase {
                 RET_CHECK(it != tokenizerOutputMap.end());
                 SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
                 embeddingsInputMap[embeddingsInputName] = it->second;
+                if (embeddingsInputName == "attention_mask") {
```

Comment:
what the impact on latency tested here?

Reply:
If the time is noticeable, some optimizations could be possible:
- in case of BS1 just check the size on 2nd dimension
- for each batch, iterate backwards and count zeros.